docs/environment/database.md

```diff
@@ -0,0 +1,147 @@
+
+> It may take some minutes for the database to be created.
+
+Find:
+
+- the "endpoint", which is the hostname of the database (this information is available only after the database creation has completed)
+
+    ℹ️ Instead of connecting via a socket, via `localhost` or an IP address, PHP will connect to MySQL via this hostname.
+- the "security group ID" (in the "VPC security groups" section), which looks like `sg-03f68e1100481622b`
+
+    ℹ️ A [security group](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html) is a firewall that restricts access to/from the VPC using "Inbound rules" and "Outbound rules".
+- the list of "subnets", which look like `subnet-12f4130e` (there are several subnets)
+
+    ℹ️ An AWS region is divided in [availability zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html) (different data centers): there is usually one subnet per availability zone.
+
+Put these information in `template.yaml` in your function configuration:
+
+```yaml
+Resources:
+    MyFunction:
+        ...
+        Properties:
+            ...
+            VpcConfig:
+                SecurityGroupIds:
+                    - sg-03f68e1100481622b
+                SubnetIds:
+                    - subnet-12f4130e
+                    - subnet-c5fe33e5
+                    - subnet-11aa85dc
+                    - subnet-85dcf240
+            Policies:
+                - AWSLambdaVPCAccessExecutionRole # Allows the lambda to access the VPC
+```
+
+Now we need to authorize connections to the RDS security group (because the lambda is in the VPC but outside of this VPC group) (https://www.reddit.com/r/aws/comments/8nr8ek/lambda_rds_connection_timeout/):
+
+- open the database configuration in RDS and click the security group
+- in the "Inbound" tab click "Edit"
+- add a rule: select MySQL/Aurora (or PostgreSQL) and set a "custom" source: select the security group itself (type `sg-` and use the autocompletion)
+- save
+
+Your PHP application will be able to connect to the database through the "endpoint" you noted above.
+
+For example a PDO connection string could be:
+
+```
+mysql://user:password@dbname.e2sctvp0nqos.us-east-1.rds.amazonaws.com/dbname
+```
+
+To learn how to properly store this connection string in your configuration head over to the ["Secrets" section of the Variables documentation](/docs/environment/variables.md#secrets).
+
+### Accessing the internet
+
+By default, a lambda in a VPC can only access resources inside the VPC. It cannot access the internet (for example external APIs) or other AWS services.
+
+To enable external access you will need to create a NAT Gateway in the VPC. Watch out, a NAT Gateway will increase the costs as they start at 27$ per month. An alternative to NAT Gateways is to split the work done by a lambda in 2 lambdas: one in the VPC that accesses the database and one outside that accesses the external API.
```

Comment:
> one in the VPC that accesses the database and one outside that accesses the external API.

this might need explaining / documenting with a how-to. right now i'm thinking "but if a function inside a VPC can't access other AWS services, how can i make it talk to another lambda function _outside_ of the VPC!?"

Reply:
Right that needs more details. TBH I wrote something similar to what's in the [serverless framework documentation](https://serverless.com/framework/docs/providers/aws/guide/functions/#vpc-configuration) out of laziness, but we can do better.

What you mention indeed won't work, but the other way around can work: a lambda function outside the VPC can do some work and then invoke a lambda *inside* to persist some data. The lambda that persists the data could as well be invoked by SQS *I assume* (but I haven't verified that).

Reply:
I have edited the paragraph. There aren't much more details for the moment but at least it's less confusing: it's more clear that the alternative solution is a solution "when possible".